src/Dapr.Actors/ActorReentrancyContextAccessor.cs

```diff
@@ -0,0 +1,48 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System.Threading;
+
+namespace Dapr.Actors
+{
+    /// <summary>
+    /// Accessor for the reentrancy context. This provides the necessary ID to continue a reentrant request
+    /// across actor invocations.
+    /// </summary>
+    public class ActorReentrancyContextAccessor
```

Comment:
This class should be `internal` and `static` - we don't want application code access this magic infrastructure.